test/e2e/pkg/dependency_test.go

```diff
@@ -0,0 +1,162 @@
+// +build e2e
+
+/*
+Copyright 2020 The Crossplane Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package pkg
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	"github.com/pkg/errors"
+	corev1 "k8s.io/api/core/v1"
+	kerrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/utils/pointer"
+	ctrl "sigs.k8s.io/controller-runtime"
+
+	"github.com/crossplane/crossplane/apis/pkg/v1beta1"
+	typedclient "github.com/crossplane/crossplane/internal/client/clientset/versioned/typed/pkg/v1beta1"
+)
+
+func TestDependencies(t *testing.T) {
+	cases := map[string]struct {
+		reason string
+		body   func() error
+	}{
+		"ResolveDependencies": {
+			reason: "Should successfully resolve dependencies for a package.",
+			body: func() error {
+				ctx := context.Background()
+				c := typedclient.NewForConfigOrDie(ctrl.GetConfigOrDie())
+				cr := &v1beta1.Configuration{
+					ObjectMeta: metav1.ObjectMeta{
+						Name: "getting-started-with-aws",
+					},
+					Spec: v1beta1.ConfigurationSpec{
+						PackageSpec: v1beta1.PackageSpec{
+							Package: "registry.upbound.io/xp/getting-started-with-aws:latest",
```

Comment:
Nit: Should we make these packages and names constants?

Reply:
@negz sounds good to me 👍 I think these tests will likely have common patterns moved out to a framework as we continue to add more e2e coverage.